app/presenters/uniprot_mapping_tsv_presenter.rb

```diff
@@ -0,0 +1,34 @@
+class UniprotMappingTsvPresenter
+    def self.objects
+      Gene.joins(variants: :evidence_items).where("variants.evidence_items.status!='rejected'").distinct
+    end
+  
+    def self.headers
+      [
+        'civic_name',
+        'uniprot_name',
+        'gene_overview'
+      ]
+    end 
+  
+    def self.row_from_object(gene)
```

Comment:
I'd rename to `rows_from_object()` and do something along these lines (haven't tested it, just off the top of my head):
```
swissprot_names = Array(Scrapers::MyGeneInfo.get_swissprot_name(gene))
formatted_overview = formatted_overview_col(gene)
swisprot_names.map do |swissprot_name|
  if name == 'N/A'
    nil
  else 
    [gene.name, swissprot_name, formatted_overview]
  end
end.compact
```

That way you have a list of rows for your TSV, `compact` will remove the `nil`s and the code that actually writes the TSV can just be a simple iteration over genes, calling this, and then writing a row for each item this returns.